pkg/commands/componentarchive/remote/remote_test.go

```diff
@@ -138,12 +139,101 @@ var _ = Describe("Remote", func() {
 		showOpts.ComponentName = "example.com/component"
 		showOpts.Version = "v6.6.6"
 
-		res := remote.NewGetCommand(ctx)
-		Expect(res)
-
 		Expect(showOpts.Run(ctx, logr.Discard(), testdataFs)).To(HaveOccurred())
 	})
 
+	It("should copy a component descriptor and its blobs from the source repository to the target repository.", func() {
+		baseFs, err := projectionfs.New(osfs.New(), "../")
+		Expect(err).ToNot(HaveOccurred())
+		testdataFs = layerfs.New(memoryfs.New(), baseFs)
+		ctx := context.Background()
+
+		cf, err := testenv.GetConfigFileBytes()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(vfs.WriteFile(testdataFs, "/auth.json", cf, os.ModePerm))
+
+		cd := &cdv2.ComponentDescriptor{}
+		cd.Name = "example.com/component"
+		cd.Version = "v0.0.0"
+		cd.Provider = cdv2.InternalProvider
+		Expect(cdv2.InjectRepositoryContext(cd, cdv2.NewOCIRegistryRepository(srcRepoCtxURL, "")))
+
+		blobContent := "blob test\n"
+
+		pushOpts := &remote.PushOptions{
+			OciOptions: options.Options{
+				AllowPlainHttp:     false,
+				RegistryConfigPath: "/auth.json",
+			},
+		}
+		pushOpts.ComponentArchivePath = "./testdata/01-ca-blob"
+		pushOpts.BaseUrl = srcRepoCtxURL
+		Expect(pushOpts.Run(ctx, logr.Discard(), testdataFs)).To(Succeed())
+
+		repos, err := client.ListRepositories(ctx, srcRepoCtxURL)
+		Expect(err).ToNot(HaveOccurred())
+
+		srcOCIRef, err := components.OCIRef(cdv2.NewOCIRegistryRepository(srcRepoCtxURL, ""), cd.Name, cd.Version)
+		expectedRef := srcRepoCtxURL + "/component-descriptors/" + cd.Name
+		Expect(err).ToNot(HaveOccurred())
+		Expect(repos).To(ContainElement(Equal(expectedRef)))
+
+		manifest, err := client.GetManifest(ctx, srcOCIRef)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(manifest.Layers).To(HaveLen(2))
+		Expect(manifest.Layers[0].MediaType).To(Equal(cdoci.ComponentDescriptorTarMimeType),
+			"Expect that the first layer contains the component descriptor")
+
+		compResolver := cdoci.NewResolver(client)
+		sourceComp, err := compResolver.Resolve(ctx, cdv2.NewOCIRegistryRepository(srcRepoCtxURL, ""), cd.Name, cd.Version)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(sourceComp.Name).To(Equal(cd.Name))
+		Expect(sourceComp.Resources[len(sourceComp.Resources)-1].Access.Type).To(Equal("localOciBlob"), "Expect that the localFilesystem blob has been correctly converted to a localOciBlob")
```

Comment:
shouldn't this always be the first element?
```suggestion
		Expect(sourceComp.Resources[0].Access.Type).To(Equal("localOciBlob"), "Expect that the localFilesystem blob has been correctly converted to a localOciBlob")
```